### PR TITLE
Fix some of the CppCheck warnings.

### PR DESCRIPTION
**Pull request by TomFischer**

### CODE REVIEW

File: GeoLib/OctTree-impl.h [hunk 1]
```diff
@@ -128,13 +128,16 @@ void OctTree<POINT, MAX_POINTS>::getPointsInRange(T const& min, T const& max,
...
-        }
+        std::copy_if(_pnts.begin(), _pnts.end(), std::back_inserter(pnts),
+                     [&min, &max](auto const* p) {
+                         if (min[0] <= (*p)[0] && (*p)[0] < max[0] &&
+                             min[1] <= (*p)[1] && (*p)[1] < max[1] &&
+                             min[2] <= (*p)[2] && (*p)[2] < max[2])
+                         {
+                             return true;
+                         }
+                         return false;
```
Comment (endJunction) [inline GeoLib/OctTree-impl.h:139]:
```suggestion
                         return (min[0] <= (*p)[0] && (*p)[0] < max[0] &&
                             min[1] <= (*p)[1] && (*p)[1] < max[1] &&
                             min[2] <= (*p)[2] && (*p)[2] < max[2]);
```

File: GeoLib/IO/XmlIO/Qt/XmlStnInterface.cpp [hunk 2]
```diff
@@ -250,8 +249,8 @@ bool XmlStnInterface::write()
     root.setAttribute( "xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance" );
 
     const std::vector<GeoLib::Point*>* stations (_geo_objs.getStationVec(_exportName));
-    bool isBorehole = static_cast<GeoLib::Station*>((*stations)[0])->type() ==
-                      GeoLib::Station::StationType::BOREHOLE;
+    bool is_borehole = static_cast<GeoLib::Station*>((*stations)[0])->type() ==
```
Comment (endJunction) [inline GeoLib/IO/XmlIO/Qt/XmlStnInterface.cpp:252]: not sure, if modified elsewhere.
```suggestion
    bool const is_borehole = static_cast<GeoLib::Station*>((*stations)[0])->type() ==
```

File: GeoLib/GEOObjects.h [hunk 3]
```diff
@@ -339,8 +339,9 @@ class GEOObjects final
      * @param pnt_offsets offsets in the merged vector storing the points
      * @return true, if merging the points succeeded, else false
      */
-    bool mergePoints(std::vector<std::string> const & geo_names, std::string & merged_geo_name,
-            std::vector<std::size_t> &pnt_offsets);
+    bool mergePoints(std::vector<std::string> const& geo_names,
+                     std::string& merged_geo_name,
```
Comment (endJunction) [inline GeoLib/GEOObjects.h:343]: in mergeSurfaces and polylines, this is const&. But could be modifying, don't know.
```suggestion
                     std::string const& merged_geo_name,
```

File: Applications/DataExplorer/VtkVis/VtkRaster.cpp [hunk 4]
```diff
@@ -309,19 +309,18 @@ std::string VtkRaster::findWorldFile(std::string const& filename)
...
 
-    std::vector<std::string> const supported_extensions =
-    { ".pgw", ".pngw", ".pgwx",
-      ".jgw", ".jpgw", ".jgwx",
-      ".tfw", ".tifw", ".tfwx",
-      ".bpw", ".bmpw", ".bpwx",
-      ".wld" };
-
-    for (auto& ext : supported_extensions)
+    std::array<std::string, 13> const supported_extensions = {
```
Comment (endJunction) [inline Applications/DataExplorer/VtkVis/VtkRaster.cpp:312]: style: It is no longer necessary to give the template arguments. Though the type changes to `std::array<char*, 13>`, which is good, but maybe expected somewhere else to be a string container.
```suggestion
    constexpr std::array supported_extensions = {
```

File: Applications/DataExplorer/VtkVis/NetCdfConfigureDialog.cpp [hunk 5]
```diff
@@ -94,9 +94,9 @@ void NetCdfConfigureDialog::on_comboBoxDim1_currentIndexChanged(int id)
 //set up y-axis/lon
 void NetCdfConfigureDialog::on_comboBoxDim2_currentIndexChanged(int id)
 {
+    Q_UNUSED(id);
     if (_currentVar.getDimCount() > 1)
     {
-        if (id == -1) id = 0;
```
Comment (endJunction) [inline Applications/DataExplorer/VtkVis/NetCdfConfigureDialog.cpp:99]: style: could be directly commented in the arguments list.
issue: same unneeded assignment above in `on_comboBoxDim1_currentIndexChanged`.

File: BaseLib/MemWatch.cpp [hunk 6]
```diff
@@ -70,20 +70,5 @@ unsigned long MemWatch::getVirtMemUsage ()
         return _vmem_size;
 }
 
-unsigned long MemWatch::getResMemUsage () {
-        updateMemUsage ();
-        return _rmem_size;
```
Comment (endJunction) [inline BaseLib/MemWatch.cpp:75]: Warning by gcc/clang: The `_rmem_size`, `_smem_size`, and `_cmem_size` unused.

File: GeoLib/GEOObjects.h (hunk 3 above)
Comment (TomFischer) [reply GeoLib/GEOObjects.h:343]: `mergePoints` calls `addPointVec` which is allowed to modify the string.

File: GeoLib/Polygon.cpp [hunk 7]
```diff
@@ -213,30 +213,33 @@ bool Polygon::containsSegment(GeoLib::LineSegment const& segment) const
...
     GeoLib::Point s;
-    for (auto polygon_seg : *this) {
-        for (auto polyline_seg : ply) {
-            if (GeoLib::lineSegmentIntersect(polyline_seg, polygon_seg, s)) {
-                return true;
-            }
+    for (auto polygon_seg : *this)
+    {
+        if (std::any_of(ply.begin(), ply.end(),
+                        [&polygon_seg, &s](auto const& polyline_seg) {
```
Comment (endJunction) [inline GeoLib/Polygon.cpp:237]: #3006 is related to the change from `auto` to `auto const&`, is my guess.

**Pull request merged**